src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.spec.ts

```diff
@@ -0,0 +1,263 @@
+import { RouterTestingModule } from '@angular/router/testing';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { mockVehicleTechnicalRecord } from '@mocks/mock-vehicle-technical-record.mock';
+import { VehicleTechRecordModel, VehicleTypes } from '@models/vehicle-tech-record.model';
+import { provideMockActions } from '@ngrx/effects/testing';
+import { Action } from '@ngrx/store';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { initialAppState } from '@store/index';
+import { updateTechRecordsSuccess } from '@store/technical-records';
+import { of, ReplaySubject } from 'rxjs';
+import { AmendVrmComponent } from './tech-record-amend-vrm.component';
+import { SharedModule } from '@shared/shared.module';
+import { DynamicFormsModule } from '@forms/dynamic-forms.module';
+import { ReactiveFormsModule } from '@angular/forms';
+import { CustomFormGroup, FormNodeEditTypes, FormNodeTypes } from '@forms/services/dynamic-form.types';
+
+const mockTechRecordService = {
+  editableTechRecord$: of({}),
+  selectedVehicleTechRecord$: of({}),
+  viewableTechRecord$: jest.fn(),
+  updateEditingTechRecord: jest.fn(),
+  isUnique: jest.fn()
+};
+
+const mockDynamicFormService = {
+  createForm: jest.fn()
+};
+
+describe('TechRecordChangeVrmComponent', () => {
+  let actions$ = new ReplaySubject<Action>();
+  let component: AmendVrmComponent;
+  let errorService: GlobalErrorService;
+  let expectedVehicle = {} as VehicleTechRecordModel;
+  let fixture: ComponentFixture<AmendVrmComponent>;
+  let route: ActivatedRoute;
+  let router: Router;
+  let store: MockStore;
+  let technicalRecordService: TechnicalRecordService;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [AmendVrmComponent],
+      providers: [
+        GlobalErrorService,
+        provideMockActions(() => actions$),
+        provideMockStore({ initialState: initialAppState }),
+        { provide: ActivatedRoute, useValue: { params: of([{ id: 1 }]) } },
+        { provide: DynamicFormService, useValue: mockDynamicFormService },
+        { provide: TechnicalRecordService, useValue: mockTechRecordService }
+      ],
+      imports: [RouterTestingModule, SharedModule, ReactiveFormsModule, DynamicFormsModule]
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AmendVrmComponent);
+    errorService = TestBed.inject(GlobalErrorService);
+    route = TestBed.inject(ActivatedRoute);
+    router = TestBed.inject(Router);
+    store = TestBed.inject(MockStore);
+    technicalRecordService = TestBed.inject(TechnicalRecordService);
+    component = fixture.componentInstance;
+    expectedVehicle = mockVehicleTechnicalRecord(VehicleTypes.PSV);
```

Comment:
The whole mockVehicle object is somewhat unnecessary for the tests below. Even if you did want it, it's worth putting in a separate `beforeEach` in the `describe` block that needs it